clients/banking/src/components/MerchantProfilePaymentArea.tsx

```diff
@@ -0,0 +1,435 @@
+
+      <Box alignItems="center">
+        <LakeButton
+          color="current"
+          onPress={handleAction}
+          disabled={actionType === "waitingForReview"}
+        >
+          {buttonText}
+        </LakeButton>
+      </Box>
+    </Box>
+  );
+};
+
+type Props = {
+  params: GetRouteParams<"AccountMerchantsProfilePaymentsArea">;
+  large: boolean;
+};
+
+export const MerchantProfilePaymentArea = ({ params, large }: Props) => {
+  const { merchantProfileId, accountMembershipId } = params;
+
+  const { canCreateMerchantPaymentLinks } = usePermissions();
+
+  const route = Router.useRoute([
+    "AccountMerchantsProfilePaymentsList",
+    "AccountMerchantsProfilePaymentsDetails",
+  ]);
+
+  const filters: MerchantPaymentFilters = useMemo(() => {
+    return {
+      paymentMethod: isNotNullish(params.paymentMethod)
+        ? Array.filterMap(params.paymentMethod, item =>
+            match(item)
+              .with("Card", "Check", "DirectDebit", value => Option.Some(value))
+              .otherwise(() => Option.None()),
+          )
+        : undefined,
+      status: isNotNullish(params.status)
+        ? Array.filterMap(params.status, item =>
+            match(item)
+              .with("Authorized", "Captured", "Initiated", "Rejected", item => Option.Some(item))
+              .otherwise(() => Option.None()),
+          )
+        : undefined,
+    } as const;
+  }, [params.paymentMethod, params.status]);
+
+  const search = nullishOrEmptyToUndefined(params.search);
+  const hasSearch =
+    isNotNullish(search) || isNotNullish(params.paymentMethod) || isNotNullish(params.status);
+
+  const [data, { isLoading, reload, setVariables }] = useQuery(MerchantPaymentsDocument, {
+    merchantProfileId,
+    first: PER_PAGE,
+    filters: {
+      status: filters.status,
+      search,
+      paymentMethod: filters.paymentMethod?.reduce<MerchantPaymentMethodType[]>((acc, item) => {
+        if (item === "DirectDebit") {
+          acc.push(
+            "SepaDirectDebitB2b",
+            "SepaDirectDebitCore",
+            "InternalDirectDebitStandard",
+            "InternalDirectDebitB2b",
+          );
+        } else {
+          acc.push(item);
+        }
+
+        return acc;
+      }, []),
+    },
+  });
+
+  const panelRef = useRef<FocusTrapRef | null>(null);
+
+  const onActiveRowChange = useCallback(
+    (element: HTMLElement) => panelRef.current?.setInitiallyFocusedElement(element),
+    [],
+  );
+
+  const getRowLink = useCallback(
+    ({ item }: LinkConfig<MerchantPaymentFragment, undefined>) => (
+      <Link
+        to={Router.AccountMerchantsProfilePaymentsDetails({
+          accountMembershipId,
+          merchantProfileId,
+          paymentId: item.id,
+        })}
+      />
+    ),
+    [accountMembershipId, merchantProfileId],
+  );
+
+  const activePaymentLinkId =
+    route?.name === "AccountMerchantsProfilePaymentsDetails"
+      ? (route.params.paymentId ?? null)
+      : null;
```

Comment:
nit, more readable:

```ts
const activePaymentLinkId = match(route)
  .with({ name: "AccountMerchantsProfilePaymentsDetails" }, ({ params }) => params.paymentId)
  .otherwise(() => null);
```